src/ad-bidders/a9/index.ts

```diff
@@ -90,6 +94,18 @@ export class A9Provider extends BidderProvider {
 		}
 	}
 
+	private invalidateSlotTargeting(params: Dictionary) {
+		const expirationDate = Date.parse(params.amznExpirationDate);
+		const currentDate = new Date().getTime();
+
+		if (expirationDate < currentDate) {
+			delete params.amznExpirationDate;
+			delete params.amznbid;
+			delete params.amzniid;
+			delete params.amznp;
```

Comment:
What do you think about:
```suggestion
			// adSlot needed
			const slotAlias = this.getSlotAlias(adSlot.getSlotName());

			delete this.bids[slotAlias];
			this.targetingKeys.forEach((key: string) => {
				context.remove(`slots.${adSlot.getSlotName()}.targeting.${key}`);
			});
```